File: VB/DXSample/ViewModel/MainViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Mvvm.DataAnnotations
Imports System.Linq

Namespace DXSample.ViewModel

    <POCOViewModel>
    Public Class MainViewModel

        Protected ReadOnly Property DocumentManagerService As IDocumentManagerService
            Get
                Return GetService(Of IDocumentManagerService)()
            End Get
        End Property

        Public Sub CreateDocument()
            Dim doc As IDocument = DocumentManagerService.CreateDocument("TabView", ViewModelSource.Create(Function() New TabViewModel()))
            doc.Id = String.Format("DocId_{0}", DocumentManagerService.Documents.Count(Of IDocument)())
            doc.Title = String.Format("Item {0}", DocumentManagerService.Documents.Count(Of IDocument)())
            doc.Show()
        End Sub

        Public Sub CloseDocument()
            DocumentManagerService.ActiveDocument.Close()
        End Sub
    End Class
End Namespace

File: CS/DXSample/ViewModel/MainViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;
using DevExpress.Mvvm.DataAnnotations;
using System;
using System.Linq;

namespace DXSample.ViewModel {
    [POCOViewModel]
    public class MainViewModel {
        protected IDocumentManagerService DocumentManagerService { get { return this.GetService<IDocumentManagerService>(); } }

        public void CreateDocument() {
            IDocument doc = DocumentManagerService.CreateDocument("TabView", ViewModelSource.Create(()=> new TabViewModel()));
            doc.Id = String.Format("DocId_{0}", DocumentManagerService.Documents.Count());
            doc.Title = String.Format("Item {0}", DocumentManagerService.Documents.Count());
            doc.Show();
        }

        public void CloseDocument() {
            DocumentManagerService.ActiveDocument.Close();
        }
    }
}
